### DISCUSSION TITLE
Does Budibase form supports AJAX?

### DISCUSSION THREAD

**Discussion by willteoh** [Budibase/budibase#6102] [General]
Use case is fields dependency, example:

1. User enters a username, without having to submit the form to the next page, it triggers an API call to the DB to check for existing usernames for validation and return the result.
2. User selects a dropdown, and depending on the selected value, it is used as a query parameter of the API call to retrieve a some data and return the result. Based on the result, the system decides to show/hide the other fields.

**Comment by melohagan** [comment]
Hi @willteoh 

This should be doable. Here's some docs pages that may be useful:

- https://docs.budibase.com/docs/actions#on-change
- https://docs.budibase.com/docs/referencing (Execute Query)
- https://docs.budibase.com/docs/conditions
